contributing/topics/guide-typed-model.md

```diff
@@ -0,0 +1 @@
+TODO: guide for adopting model objects linked to from the Data Source/Resource code
```

Comment:
Is this something we want to keep in here for ourselves to know what to add later?